### PR TITLE
feat: traffic shaping

### PR DESCRIPTION
**Pull request by royalpinto007**
Documentation for Traffic Shaping.

### CODE REVIEW

File: NetworkConfigurations/TrafficControlAndPolicingSubCategories/TrafficShaping.md [hunk 1]
```diff
@@ -0,0 +1,417 @@
+---
+title: Traffic Shaping
+grand_parent: Network Configuration
+parent: Traffic Control and Policing
+nav_order: 3
+layout: default
+---
+
+# Traffic Shaping
```
Comment (taskin0003) [inline NetworkConfigurations/TrafficControlAndPolicingSubCategories/TrafficShaping.md:9]: How did we verify the configurations below are functional and perform as intended?

**Review by taskin0003 — CHANGES_REQUESTED** [review]

File: NetworkConfigurations/TrafficControlAndPolicingSubCategories/TrafficShaping.md (hunk 1 above)
Comment (royalpinto007) [reply NetworkConfigurations/TrafficControlAndPolicingSubCategories/TrafficShaping.md:9]: Verified using `iperf` utility and listed the outputs at https://github.com/dentproject/docs.dent.dev/pull/41/commits/cff727ccb266a2d909d61fbbbdbbdeb27d98b02e.

**Review by taraschornyiplv — DISMISSED** [review]

File: NetworkConfigurations/TrafficControlAndPolicingSubCategories/TrafficShaping.md [hunk 2]
```diff
@@ -0,0 +1,506 @@
...
+------------------------------------------------------------
+Client connecting to 127.0.0.1, TCP port 5001
+TCP window size: 2.50 MByte (default)
+------------------------------------------------------------
+[  3] local 127.0.0.1 port 52146 connected with 127.0.0.1 port 5001
+[ ID] Interval       Transfer     Bandwidth
+[  3]  0.0-60.0 sec  57.2 MBytes  8.00 Mbits/sec
+```
+
+Given that the configuration has a bandwidth limit of 10 Mbit/s, setting the iperf bandwidth slightly below this limit to observe the effect of traffic shaping is appropriate. In this case, the achieved bandwidth aligns with the specified bandwidth, indicating that the traffic shaping configuration is working as expected, effectively limiting the bandwidth to the configured rate of 8 Mbps.
```
Comment (taskin0003) [inline NetworkConfigurations/TrafficControlAndPolicingSubCategories/TrafficShaping.md:191]: Bandwidth is limited to 10Mb/s. If we send 8Mb/s we should expect 8Mb/s, which we see. Shouldn't the real bandwidth limiting be observed by sending more than 10Mb/s through iperf and observe 10Mb/s?

I would keep the 8Mb/s example to show traffic is not shaped. But add sending something like 15Mb/s and observe 10Mb/s.

File: NetworkConfigurations/TrafficControlAndPolicingSubCategories/TrafficShaping.md [hunk 3]
```diff
@@ -0,0 +1,506 @@
...
+------------------------------------------------------------
+Client connecting to 127.0.0.1, TCP port 5001
+TCP window size: 2.50 MByte (default)
+------------------------------------------------------------
+[  3] local 127.0.0.1 port 51982 connected with 127.0.0.1 port 5001
+[ ID] Interval       Transfer     Bandwidth
+[  3]  0.0-60.0 sec  2.44 GBytes   350 Mbits/sec
+```
+
+The iperf test ran for 60 seconds, during which approximately 2.44 Gigabytes of data were transferred. The average bandwidth achieved during this test is reported as 350 Mbits/sec, which aligns with the specified bandwidth of 350 Mbps in the iperf command (-b 350M). This indicates that the traffic shaping configuration is allowing traffic to flow at the desired rate, limiting it to approximately 350 Mbps as specified.
```
Comment (taskin0003) [inline NetworkConfigurations/TrafficControlAndPolicingSubCategories/TrafficShaping.md:260]: You've configured iperf to send 350 Mb/s send measured that rate. How does it prove the bandwidth is indeed limited? Send something like 400Mb/s and observe 350Mb/s.

File: NetworkConfigurations/TrafficControlAndPolicingSubCategories/TrafficShaping.md [hunk 4]
```diff
@@ -0,0 +1,506 @@
...
+Client connecting to 127.0.0.1, TCP port 5001
+TCP window size: 2.50 MByte (default)
+------------------------------------------------------------
+[  3] local 127.0.0.1 port 51824 connected with 127.0.0.1 port 5001
+[ ID] Interval       Transfer     Bandwidth
+[  3]  0.0-60.0 sec  2.10 GBytes   300 Mbits/sec
+```
+
+The test results indicate that the achieved bandwidth was indeed around 300 Mbits/sec, which is expected given that the traffic shaping is configured to limit the rate to 400 Mbit/s.
+
```
Comment (taskin0003) [inline NetworkConfigurations/TrafficControlAndPolicingSubCategories/TrafficShaping.md:414]: Again, I do not believe this example proves you've limited the bandwidth. The limit is set to 400Mb/s, and you've sent 300Mb/s. Shouldn't you expect 300Mb/s?

File: NetworkConfigurations/TrafficControlAndPolicingSubCategories/TrafficShaping.md [hunk 5]
```diff
@@ -0,0 +1,506 @@
...
+Client connecting to 127.0.0.1, TCP port 5001
+TCP window size: 2.50 MByte (default)
+------------------------------------------------------------
+[  3] local 127.0.0.1 port 51748 connected with 127.0.0.1 port 5001
+[ ID] Interval       Transfer     Bandwidth
+[  3]  0.0-10.0 sec  11.9 MBytes  10.0 Mbits/sec
+```
+
+The test was executed successfully, showing a transfer of approximately 11.9 Megabytes of data at an average bandwidth of 10.0 Mbps, which aligns with the configured bandwidth of 10 Mbps.
+
```
Comment (taskin0003) [inline NetworkConfigurations/TrafficControlAndPolicingSubCategories/TrafficShaping.md:499]: Similar comments as the previous three. I do not know if these examples are valid.

**Review by taskin0003 — CHANGES_REQUESTED** [review]

**Review by taraschornyiplv — APPROVED** [review]

**Review by taskin0003 — APPROVED** [review]

**Pull request merged**